lib/misc.py

```diff
@@ -42,7 +42,11 @@ def delete(cli):
 def list_(cli):
    ch.dependencies_check()
    imgdir = ch.storage.unpack_base
-   imgs = ch.ossafe(os.listdir, "can't list directory: %s" % imgdir, imgdir)
+   if (not os.path.isdir(ch.storage.root)):
+      ch.FATAL("does not exist: %s" % ch.storage.root)
+   if (not ch.storage.valid_p()):
+       ch.FATAL("doesn't contain subdirectory img: %s" % ch.storage.root)
+   imgs = ch.ossafe(os.listdir, "can't list directory: %s" % ch.storage.root, imgdir)
```

Comment:
A few thoughts:
1. If the storage root doesn't exist, why not create it and then say there are no images?
2. Do we need to exit with a failure? For example:
```
$ ch-image list
No images in storage.
$ echo $?
0
```